server/plugin.go

```diff
@@ -113,13 +120,42 @@ func (p *Plugin) OnActivate() error {
 		return mutateSampleLabels(l, target, *p.configuration.HonorTimestamps, []*relabel.Config{})
 	}
 
+	ticker := time.Tick(time.Duration(scrapeInterval) * time.Second)
```

Comment:
I'd recommend to use `time.Ticker` instead and explicitly stop it when no longer needed.